src/Microsoft.AspNetCore.JsonPatch/Adapters/ObjectAdapter.cs

```diff
@@ -11,24 +11,43 @@ namespace Microsoft.AspNetCore.JsonPatch.Adapters
     /// <inheritdoc />
     public class ObjectAdapter : IObjectAdapterWithTest
     {
+        /// <summary>
+        /// Initializes a new instance of <see cref="ObjectAdapter"/> using the default AdapterFactory.
+        /// </summary>
+        /// <param name="contractResolver">The <see cref="IContractResolver"/>.</param>
+        /// <param name="logErrorAction">The <see cref="Action"/> for logging <see cref="JsonPatchError"/>.</param>
+        public ObjectAdapter(
+            IContractResolver contractResolver,
+            Action<JsonPatchError> logErrorAction):this(contractResolver, logErrorAction, new AdapterFactory())
+        {
+        }
+
         /// <summary>
         /// Initializes a new instance of <see cref="ObjectAdapter"/>.
         /// </summary>
         /// <param name="contractResolver">The <see cref="IContractResolver"/>.</param>
         /// <param name="logErrorAction">The <see cref="Action"/> for logging <see cref="JsonPatchError"/>.</param>
+        /// <param name="adapterFactory">The <see cref="IAdapterFactory"/> to use when creating adaptors.</param>
         public ObjectAdapter(
             IContractResolver contractResolver,
-            Action<JsonPatchError> logErrorAction)
+            Action<JsonPatchError> logErrorAction,
+            IAdapterFactory adapterFactory)
         {
             ContractResolver = contractResolver ?? throw new ArgumentNullException(nameof(contractResolver));
             LogErrorAction = logErrorAction;
+            AdapterFactory = adapterFactory;
         }
 
         /// <summary>
         /// Gets or sets the <see cref="IContractResolver"/>.
         /// </summary>
         public IContractResolver ContractResolver { get; }
 
+        /// <summary>
+        /// Gets or sets the <see cref="IAdapterFactory"/>
+        /// </summary>
+        public IAdapterFactory AdapterFactory { get; set; }
```

Comment:
How about making this readonly (removing setter)?

Reply:
Just preference I guess.  I like to have my properties externally set able in case it makes unit testing easier.  That being said we don't currently need it, so if you would prefer it to be read only, I can do that.  Again, just let me know it is an easy fix.  Thanks for the quick feedback.

Reply:
Yea, I'd prefer that it's read-only.  This makes it consistent with the other constructor parameters `contractResolver` and ``logErrorAction`.  Constructor-set properties are generally read-only in my experience.